packages/vscode-extension/src/webview/components/UrlBar.tsx

```diff
@@ -65,32 +98,34 @@ function UrlBar({ project, disabled }: UrlBarProps) {
           label: "Go back",
           side: "bottom",
         }}
-        disabled={disabled || urlList.length < 2}
+        disabled={disabled || urlHistory.length < 2}
         onClick={() => {
-          project.openNavigation(urlList[1].id);
-          // remove first item from the url list
-          setUrlList((urlList) => urlList.slice(1));
+          setUrlHistory((prevUrlHistory) => {
+            const newUrlHistory = prevUrlHistory.slice(1);
+            project.openNavigation(newUrlHistory[0]);
+            return newUrlHistory;
+          });
         }}>
         <span className="codicon codicon-arrow-left" />
       </IconButton>
       <ReloadButton project={project} disabled={disabled ?? false} />
       <IconButton
         onClick={() => {
           project.goHome();
-          setUrlList([]);
         }}
         tooltip={{
           label: "Go to main screen",
           side: "bottom",
         }}
-        disabled={disabled || urlList.length == 0}>
+        disabled={disabled || urlList.length < 1}>
         <span className="codicon codicon-home" />
       </IconButton>
       <UrlSelect
         onValueChange={(value: string) => {
           project.openNavigation(value);
         }}
-        items={urlList}
+        recentItems={recentUrlList}
```

Comment:
When changing from a preview to another URL, two route changes happen: {displayName: '', id: '{}'} and {displayName: '/', id: '/{}'}. We skip storing the first route with an empty displayName to prevent saving home URL entries twice.